src/app-layer-smtp.c

```diff
@@ -681,6 +730,11 @@ static AppLayerResult SMTPGetLine(
         input->len -= line->len;
         DEBUG_VALIDATE_BUG_ON((input->consumed + input->len) != input->orig_len);
         line->buf = input->buf + o_consumed;
+
+        if (frame != NULL) {
+            frame->len = (int64_t)line->len;
```

Comment:
Does this work for multiline responses ? Like server initial banner with capabilities...

Reply:
Should not
`alert smtp any any -> any any (frame:smtp.response_line; content:"250-PIPELINING"; content:"250-SIZE"; sid:3;)` trigger on smtp-pipelining SV test ?

Reply:
It does not. It's a per line frame. I guess a more generic `smtp.response` (like wireshark) could be added that would bundle up lines that belong in single response. Not planning to do that now though.

Reply:
Fair enough, thanks